src/unix/special.c

```diff
@@ -340,10 +340,16 @@ boolean create_special_file(const char *path, spec_file_open open, u64 size)
 
 void register_special_files(process p)
 {
+    filesystem fs = p->root_fs;
+    tuple self_exe;
     heap h = heap_locked((kernel_heaps)p->uh);
 
-    fs_status fss = filesystem_mkdirpath(p->root_fs, 0, "/proc/self", true);
+    fs_status fss = filesystem_get_node(&fs, p->cwd, "/proc/self/exe", false, false, false,
+        &self_exe, 0);
     if (fss == FS_STATUS_OK) {
+        filesystem_put_node(fs, self_exe);
+    } else {
+        filesystem_mkdirpath(p->root_fs, 0, "/proc/self", true);
```

Comment:
Nit, but might want to check that the result here is FS_STATUS_OK or FS_STATUS_EXIST and not some other error.

Reply:
OK, now the symlink is created only if filesystem_mkdirpath() returns FS_STATUS_OK or FS_STATUS_EXIST